### ISSUE TITLE
Issue with Guild Communications

### ISSUE THREAD

**Issue by croana** [ebonyfaye/ema#29]
Heya,

I'm one of those dirty people who has literally all of their characters from one faction in one giant EMA team profile.

As suggested, I turned on guild communications in order to reduce lag when the team is mounting/dismounting (which def helps, thanks.)

I've found an issue with guild communications when characters in one team are in different guilds, even if all the ONLINE chars happen to be in the same guild. Here's the steps:

0) For this example - 
Char1a, Char1b, Char1c... are all in Guild1

Char2a, Char2b, Char2c... are all in Guild2

Char1a,1b,1c,2a,2b,2c are all in the same team in the same EMA profile.

1) I enable guild communications while logged in on Char1a-c in Guild1, and send the profile to all other online chars. This is the first time I've enabled guild communications.

2) I log onto Char2a-c in Guild2. Guild communications is still on in EMA, because all these chars share the same team/profile with Char1a-c in Guild1.

3) I get MASSIVE stuttering on Char2a-c. About every 20 seconds or so the game appears to lag out completely. Weirdly, this doesn't actually seem to happen (often?) when ONLY the EMA addon is running, but as soon as I turn on literally any other addon ever, the stuttering appears.

4) I disable guild communications on Char2a. The stuttering stops on that character. I now can't send this information to Char2b and 2c, because they don't share the same communication channels anymore. I have to turn guild communications off manually on all characters in order to stop the stuttering.

5) I enable guild communications on Char2a again. The stuttering returns. If I try to reload UI or log out/in before turning it back on, I still get this issue. It seems that something in the background has locked guild communications to Guild1, and will always cause issues with Guild2 now.

The obvious current solutions to this problem are either:
- Don't use guild communications on any characters, lol
- Disable guild communications manually on any characters in Guild2 whenever you want to play them
- Create a fresh profile only for the chars in Guild2, and make sure to enable guild communications for the first time while in Guild2 on that profile.

The nicer solution would be:
- Stuttering doesn't happen when currently online characters are in the "wrong" guild for guild communications.
- If currently online characters aren't in the same guild, communications should default back to the addon communication channel.
- Guild communications should work so long as all currently online characters are in the same guild, regardless of what guild that is. (Chara1-c in Guild1 should auto uses Guild1 coms; Char2a-c in Guild2 auto uses Guild2 coms; Char1a, 2b, 2c as a mixed team auto uses addon coms because they aren't in the same guild)

Thanks! <3

**Comment by DejayRezme** [comment]
Not sure if this helps but you can also copy profiles. So making a profile per team and kicking out the rest shouldn't be that difficult.

**Comment by croana** [comment]
The problem with copying a profile is you copy everything, including the first guild you linked up with. What I did, instead, was create a fresh profile on one account's character, and keep the old profile on a 2nd account's character. I then turned off guild communication on both characters, made the char with the old profile master of the team, added both chars to a new team on the char with the fresh profile, and synced every module one by one EXCEPT the communication module. At that point I could create a fresh profile on all my other accounts, joined up in a party, added members to the team on all windows, and synced from the first char with a fresh profile. Finally, I could then turn guild communications back on, since all of these characters were in the correct OTHER guild.

Hope that makes sense.

It's def not ideal, but luckily I just have one multiboxing guild and one raiding guild on my main server. :)

The only thing that remains to be seen is what happens when I log onto a character that's on a different server? That's probably a new guild, too. It seems to me, at that point, it'll be pretty impossible to, for example, gather herbs on multiple servers at once while guild communication is turned on. I'm not really sure how many people actually do this while multiboxing? I'm super lazy and barely farm with my team as it is.
